tests/test_other.py

```diff
@@ -8337,12 +8384,15 @@ def test_single_file(self):
       ):
         continue
 
-      expect_asmjs_code = asmjs_fallback_enabled and wasm_enabled
+      expect_wasm = wasm_enabled
       expect_emterpretify_file = emterpreter_file_enabled
       expect_meminit = (meminit1_enabled and not wasm_enabled) or (wasm_enabled and asmjs_fallback_enabled)
       expect_success = not (emterpreter_file_enabled and single_file_enabled)
-      expect_wasm = wasm_enabled
-      expect_wast = debug_enabled and wasm_enabled
+      expect_asmjs_code = asmjs_fallback_enabled and wasm_enabled and not self.is_wasm_backend()
+      expect_wast = debug_enabled and wasm_enabled and not self.is_wasm_backend()
+
+      if self.is_wasm_backend() and (asmjs_fallback_enabled or emterpreter_enabled or not wasm_enabled):
```

Comment:
`asmjs_fallback_enabled` is redundant here (can only be true if `self.is_wasm_backend()` is false, which would always make this if false)